ssr/src/page/icpump/mod.rs

```diff
@@ -0,0 +1,120 @@
+use leptos::*;
+
+use crate::component::spinner::FullScreenSpinner;
+use crate::consts::ICPUMP_LISTING_PAGE_SIZE;
+use crate::utils::token::icpump::get_paginated_token_list;
+use crate::utils::token::icpump::TokenListItem;
+
+#[component]
+pub fn TokenListing(details: TokenListItem) -> impl IntoView {
+    let created_at_str = details.created_at.clone();
+    let datetime = chrono::DateTime::parse_from_rfc3339(&created_at_str).unwrap();
+    let elapsed = chrono::Utc::now()
+        .signed_duration_since(datetime)
+        .num_seconds();
+    let elapsed_str = if elapsed < 60 {
+        format!("{}s ago", elapsed)
+    } else if elapsed < 3600 {
+        format!("{}m ago", elapsed / 60)
+    } else if elapsed < 86400 {
+        format!("{}h ago", elapsed / 3600)
+    } else {
+        format!("{}d ago", elapsed / 86400)
+    };
+
+    view! {
+        <div class="max-h-[300px] overflow-hidden h-fit p-2 flex border hover:border-white gap-2 w-full  border-transparent">
+            <div class="min-w-32 relative self-start">
+            <img class="mr-4 w-32 h-auto" src={details.logo} alt={details.token_name.clone()}/>
+            </div>
+            <div class="gap-1 grid h-fit">
+                <span class="text-sm text-gray-500 font-bold">"$" {details.token_symbol}</span>
+                <span class="text-sm text-gray-500">"Name: " {details.token_name}</span>
+                <span class="text-sm text-gray-500">{details.description}</span>
+                <span class="text-xs text-gray-500">{elapsed_str}</span>
+                <span class="text-xs text-gray-500">"Created by: " {details.user_id}</span>
+            </div>
+        </div>
+    }
+}
+
+#[component]
+pub fn ICPumpListing() -> impl IntoView {
+    let page = create_rw_signal(1);
+    let token_list: RwSignal<Vec<TokenListItem>> = create_rw_signal(vec![]);
+    let end_of_list = create_rw_signal(false);
+
+    let act = create_resource(page, move |_| async move {
+        get_paginated_token_list(page.get_untracked())
+            .await
+            .unwrap()
```

Comment:
I want you to skip fetching this if its not required.
A simple check which checks the length should be enough I think?
`if token_list.with_untracked(|l| l.len()) > page * ICPUMP_LISTING_PAGE_SIZE` (along with a few edge cases for end_of_list)
Feel free to check and ensure we don't make extraneous network call